news/WildFly-s2i-19-released.adoc

```diff
@@ -0,0 +1,148 @@
+```
+* Chained build template: 
+```
+oc create -f https://raw.githubusercontent.com/wildfly/wildfly-s2i/wf-19.0/templates/wildfly-s2i-chained-build-template.yml
+```
+
+NB: If you import the image streams into your project, be sure to set the _ImageStreams Namespace_ (_IMAGE_STREAM_NAMESPACE_ argument) to your project namespace in the template. _openshift_ being the default namespace.
+
+== Clustering examples
+
+=== Configure Openshift
+
+* Allow to view all pods in the project: 
+```
+oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default
+```
+
+* Be sure to kill any WildFly application running in the _myproject_ namespace. Any running application would join the cluster.
+
+
+=== Web session sharing using the _web-clustering_ galleon layer
+
+==== Build and run the application
+
+We are provisioning a web server with support for web session sharing.
+
+* Build the application image:
+```
+oc new-app wildfly-s2i-chained-build-template -p APPLICATION_NAME=web-clustering \
+      -p GIT_REPO=https://github.com/wildfly/wildfly-s2i \
+      -p GIT_CONTEXT_DIR=examples/web-clustering \
+      -p GALLEON_PROVISION_LAYERS=web-server,web-clustering \
+      -p IMAGE_STREAM_NAMESPACE=myproject
+```
+
+* Create an application from the application image:
+
+```
+oc new-app myproject/web-clustering -e KUBERNETES_NAMESPACE=myproject -e JGROUPS_CLUSTER_PASSWORD=mypassword
+```
+
+NB: The KUBERNETES_NAMESPACE is required to see other pods in the project, otherwise the server attempts to retrieve pods from the 'default' namespace which is not the one our project is using.
+JGROUPS_CLUSTER_PASSWORD is used to authenticate servers in the cluster.
+
+* Expose the service:
+```
+oc expose svc/web-clustering
+```
+
+* Access the application route, note the user created time and session ID.
+
+* Scale the application to 2 pods: 
+```
+oc scale --replicas=2 dc web-clustering
+```
+
+* List pods: 
+```
+oc get pods
+```
+
+* Kill the oldest POD (which answered the first application request):
+```
+oc delete pod web-clustering-1-r4cx8 -n myproject
+```
+
+* Access the application again. You will notice that the displayed values are the same, meaning the web session has been shared between the 2 pods.
+
+=== EJB singleton
+
+Another aspect of the WildFly clustering features is the support for high availabilty EJB singletons. We are here running the _ha-singleton-deployment_ WildFly quickstart.
+In this example we are using the default server present in the wildfly s2i builder image; we are not provisioning a server trimmed-down with Galleon. We don't have (yet) layers
+for EJB features.
+
+* Build the application image:
+```
+oc new-app wildfly-s2i-chained-build-template -p APPLICATION_NAME=singleton \
+      -p GIT_REPO=https://github.com/wildfly/quickstart/ \
+      -p GIT_BRANCH=19.0.0.Final \
+      -p GIT_CONTEXT_DIR=ha-singleton-deployment  \
+      -p IMAGE_STREAM_NAMESPACE=myproject \
```

Comment:
as the example assumes that the image streams have been installed in `myproject` namespace, maybe we should explicitly mention that step in the installation of the image streams above?